templates/profile/hathitrust/solr6/index-release.sh.erb

```diff
@@ -0,0 +1,105 @@
+#! /bin/bash
+
+TODAY=`date +%Y-%m-%d`
+#TODAY="2019-08-31"
+
+BASE=/htsolr/<%= @solr_name %>
+SYMLINKBASE=/htsolr/serve
+SNAP=".snapshot/<%= @snapshot_name %>_${TODAY}"
+CORES="<%= @solr_cores.join(' ') %>"
+
+# check if the administrative release stop flag is set
+if [ -f ${BASE}/flags/<%= @solr_stop_flag %> ]; then
+  /bin/echo "<%= @solr_stop_flag %> flag present...skip release today"
+  exit 1
+fi
+
+# check that today's snap exists
+for s in $CORES; do
+  if [ ! -d $BASE/cores/${s}/${SNAP} ]; then
+    /bin/echo "missing today's snapshot...aborting release"
+    exit 1
+  fi
+done
+
+# check today's snap for 'busy' flag
+if [ -f $BASE/flags/${SNAP}/busy ]; then
+  /bin/echo "indexing and/or optimization ocurring...skip release today"
+  exit 1
+fi
+
+<% if not @is_primary_site %>#<% end %># check whether release happened at mirror site
```

Comment:
I would prefer structuring these all as e.g.
```bash
<% if not @is_primary_site %>
# check whether release happened at mirror site
if ! curl -A SOLR -s --retry 5 --fail https://babel.hathitrust.org/flags/web/<%= @solr_name %>-release-${TODAY} --resolve "babel.hathitrust.org:443:<%= @mirror_site_ip %>" >& /dev/null; then
  /bin/echo "Mirror site index release appears to have failed...skip release today"
  exit 1
fi
<% end %>
```
rather than repeating the check on every line and adding commented-out things in the output file.

Reply:
Yeah, this logic tortured. Pushed at update to change it.
IIRC I was trying to make the output have a consistent line count to make it easier to diff in development.

Reply:
Thanks, looks much clearer now.